articles/android-rxjava/index.md

```diff
@@ -0,0 +1,127 @@
+#### RxJava in Android
+ReactiveX has created libraries for almost all programming languages. It has also extended this to frameworks like Android. This has helped bring the functional reactive paradigm to the languages and frameworks we use. In Android, it helps bring efficiency in application performance. It also ensures a clean and maintainable code. This article will go through the basics of RxJava and how to use it in Android development.
+
+#### Prerequisites
+To follow through this article, you need :
+* Android Studio installed
+* Basic understanding of Android development using Kotlin
+* Experience with retrofit and room
+
+#### What we'll look at :
+1. [Observables](#Observables)
+    * [observable](#observable)
+    * [flowable](#flowable)
+    * [single](#single)
+    * [completable](#completable)
+
+2. [Schedulers](#Schedulers)
+    * [newThread](#newThread)
+    * [io](#io)
+    * [computation](#computation)
+    * [mainThread](#mainThread)
+
+3. [Application in Android](#Application-in-Android)
+    * [retrofit](#retrofit)
+    * [room](#room)
+
+
+#### Observables
+RxJava, like the other ReactiveX libraries, uses the observable pattern. This pattern involves an `observer` who subscribes to an `observable`. The observer then receives data emitted by the observable and performs actions on it. An observable is an object or entity that emits data. We have five types of observers in RxJava, but we will look at only four. This is because the `Maybe` observable is not encouraged in Android applications. These observables have some standard methods in their classes to help in managing the emissions. The methods are `onNext`, `onCompleted`, `onSuccess` and `onError`.
+
+- **_onNext_** - the observable invokes this method when data when it receives data. The result contains data that has it has received. It is available in observable and flowable types since they can emit data more than once.
+
+- **_onSuccess_** - as the name states, it is called when an operation is successful. It is mostly used in `single` observables since they emit data only once.
+
+- **_onCompleted_** - observables call this method once it finishes emitting data or doing a task. It is available in all observable types.
+
+- **_onError_** - this is called when an error occurs. It emits a `Throwable` through which a developer can get the error details.
+
+
+##### `observable`
+This is the primary type of observable. It emits data as it receives it. It can be data from a collection like a list or a map. It has the `onNext`, `onError`, and `onCompleted` methods. It calls the onCompleted once onNext emits that last piece of data.
```

Comment:
Add backticks to the keywords on the last line.

Reply:
done...